src/Connection.php

```diff
@@ -50,25 +50,24 @@ class Connection
 {
     /**
      * Represents an array of ints to be expanded by Doctrine SQL parsing.
+     *
+     * @deprecated Use {@see ArrayParameterType::INTEGER} instead.
      */
-    public const PARAM_INT_ARRAY = ParameterType::INTEGER + self::ARRAY_PARAM_OFFSET;
+    public const PARAM_INT_ARRAY = ArrayParameterType::INTEGER;
 
     /**
      * Represents an array of strings to be expanded by Doctrine SQL parsing.
+     *
+     * @deprecated Use {@see ArrayParameterType::STRING} instead.
      */
-    public const PARAM_STR_ARRAY = ParameterType::STRING + self::ARRAY_PARAM_OFFSET;
+    public const PARAM_STR_ARRAY = ArrayParameterType::STRING;
 
     /**
      * Represents an array of ascii strings to be expanded by Doctrine SQL parsing.
-     */
-    public const PARAM_ASCII_STR_ARRAY = ParameterType::ASCII + self::ARRAY_PARAM_OFFSET;
-
-    /**
-     * Offset by which PARAM_* constants are detected as arrays of the param type.
      *
-     * @internal Should be used only within the wrapper layer.
+     * @deprecated Use {@see ArrayParameterType::ASCII} instead.
      */
-    public const ARRAY_PARAM_OFFSET = 100;
```

Comment:
I'm fine with this removal because it's internal. How do others see this?

Reply:
Favorably

Reply:
ORM seems use it https://github.com/doctrine/dbal/issues/5837#issuecomment-1367679813 and if doctrine/dbal and doctrine/orm is not a match it will fail. So for bc reasons when I think it need to be kept if somebody is only updating doctrine/dbal and not doctrine/orm. Else doctrine/dbal 3.6.0 would require to set a conflict to all doctrine/orm versions lower than a supported version that version to avoid that older none compatible orm versino is used, setting requirement on doctrine/orm side would not work as current orm version already supports ^3.6.

Reply:
Good point. I'll restore the constant.

Reply:
#5842

Reply:
@derrabus Thank you!